flink-doris-connector/src/main/java/org/apache/doris/flink/sink/schema/SQLParserSchemaManager.java

```diff
@@ -0,0 +1,191 @@
+            Integer length = null;
+            Integer scale = null;
+            if (CollectionUtils.isNotEmpty(colDataType.getArgumentsStringList())) {
+                List<String> argumentsStringList = colDataType.getArgumentsStringList();
+                length = Integer.parseInt(argumentsStringList.get(0));
+                if (argumentsStringList.size() == 2) {
+                    scale = Integer.parseInt(argumentsStringList.get(1));
+                }
+            }
+            datatype =
+                    JsonDebeziumChangeUtils.buildDorisTypeName(
+                            sourceConnector, datatype, length, scale);
+
+            List<String> columnSpecs = columnDataType.getColumnSpecs();
+            String defaultValue = extractDefaultValue(columnSpecs);
+            String comment = extractComment(columnSpecs);
+            FieldSchema fieldSchema = new FieldSchema(columnName, datatype, defaultValue, comment);
+            String addColumnDDL = SchemaChangeHelper.buildAddColumnDDL(dorisTable, fieldSchema);
+            LOG.info("Parsed add column DDL SQL is: {}", addColumnDDL);
+            addColumnList.add(addColumnDDL);
+        }
+        return addColumnList;
+    }
+
+    private String processChangeColumnOperation(
+            AlterExpression alterExpression, String dorisTable) {
+        String columnNewName = alterExpression.getColDataTypeList().get(0).getColumnName();
+        String columnOldName = alterExpression.getColumnOldName();
+        String renameColumnDDL =
+                SchemaChangeHelper.buildRenameColumnDDL(dorisTable, columnOldName, columnNewName);
+        LOG.warn(
+                "Note: Only rename column names are supported in doris. "
+                        + "Therefore, the change syntax used here only supports the use of rename."
+                        + " Parsed change column DDL SQL is: {}",
+                renameColumnDDL);
+        return renameColumnDDL;
+    }
+
+    private String processRenameColumnOperation(
+            AlterExpression alterExpression, String dorisTable) {
+        String columnNewName = alterExpression.getColumnName();
+        String columnOldName = alterExpression.getColumnOldName();
+        String renameColumnDDL =
+                SchemaChangeHelper.buildRenameColumnDDL(dorisTable, columnOldName, columnNewName);
+        LOG.info("Parsed rename column DDL SQL is: {}", renameColumnDDL);
+        return renameColumnDDL;
+    }
+
+    private String extractDefaultValue(List<String> columnSpecs) {
+        String defaultValue = null;
+        if (columnSpecs.contains("default")) {
+            int defaultIndex = columnSpecs.indexOf("default");
+            defaultValue = removeQuotes(columnSpecs.get(defaultIndex + 1));
+        } else if (columnSpecs.contains("DEFAULT")) {
+            int defaultIndex = columnSpecs.indexOf("DEFAULT");
+            defaultValue = removeQuotes(columnSpecs.get(defaultIndex + 1));
```

Comment:
Will there be any problem with this judgment?

Reply:
For example, in this example, there is no special method to obtain default and comment, only through the columnSpecs collection.
In order to avoid NPE, I also added some prevention methods.
<img width="1033" alt="image" src="https://github.com/user-attachments/assets/cbde0ff8-6382-45a8-8d96-dd091a11dead">